Security/Voter/SimplifiedCoreVoter.php

```diff
@@ -56,11 +57,15 @@ public function vote(TokenInterface $token, $object, array $attributes)
             $attribute = substr($attribute, strlen(static::EZ_ROLE_PREFIX));
             list($module, $function) = explode(':', $attribute);
             $attributeObject = new AuthorizationAttribute($module, $function);
-            if ($object instanceof ValueObject) {
-                $attributeObject->limitations = ['valueObject' => $object];
-                return $this->valueObjectVoter->vote($token, $object, [$attributeObject]);
-            } else {
-                return $this->coreVoter->vote($token, $object, [$attributeObject]);
+            try {
+                if ($object instanceof ValueObject) {
+                    $attributeObject->limitations = ['valueObject' => $object];
+                    return $this->valueObjectVoter->vote($token, $object, [$attributeObject]);
+                } else {
+                    return $this->coreVoter->vote($token, $object, [$attributeObject]);
+                }
+            } catch (InvalidArgumentException $e) {
```

Comment:
AFAICS in referenced PR in kernel, it's an `InvalidArgumentValue` which is thrown, shouldn't we catch it instead?

Reply:
`InvalidArgumentValue` is a concrete implementation, while this one is an interface. I'll let you decide :)

Reply:
Ah OK, I thought it was the implementation, I misread the `use` statement 😊